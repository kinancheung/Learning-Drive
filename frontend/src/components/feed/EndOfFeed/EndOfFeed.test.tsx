import { render, screen } from '@testing-library/react';

import { EndOfFeed } from '.';

describe('EndOfFeed', () => {
  it('renders without error', () => {
    const view = render(<EndOfFeed />);

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders the message given', () => {
    const message = 'This is going to be shown in the component';

    render(<EndOfFeed message={message} />);

    const textDisplay = screen.getByTestId('end-of-feed-message');

    expect(textDisplay).toHaveTextContent(message);
  });
});
