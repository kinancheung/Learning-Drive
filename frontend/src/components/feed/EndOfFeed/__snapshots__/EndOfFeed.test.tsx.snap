// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndOfFeed renders without error 1`] = `
<DocumentFragment>
  <div
    class="css-1u06v6n-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k9fvci-MuiSvgIcon-root"
      data-testid="SadBoxIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 48 48"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M45 18.5V42.5H3V18.5M45 18.5L38 10M45 18.5H38M3 18.5L10 10M3 18.5H10M10 10H38M10 10V18.5M38 10V18.5M10 18.5H18C18 21.2614 20.4624 23.5 23.5 23.5C26.5376 23.5 29 21.2614 29 18.5H38"
          stroke="currentColor"
          stroke-linejoin="round"
          stroke-width="3"
        />
        <path
          d="M13 26.5C13 27.8807 11.8807 28.5 10.5 28.5C9.11929 28.5 8 27.8807 8 26.5C8 25.1193 9.11929 24 10.5 24C11.8807 24 13 25.1193 13 26.5Z"
          fill="currentColor"
        />
        <path
          d="M40 26.5C40 27.8807 38.8807 28.5 37.5 28.5C36.1193 28.5 35 27.8807 35 26.5C35 25.1193 36.1193 24 37.5 24C38.8807 24 40 25.1193 40 26.5Z"
          fill="currentColor"
        />
        <path
          d="M16 35.5C16 35.5 19 34 24 34C29 34 32 35.5 32 35.5"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="3"
        />
      </svg>
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 css-19f7wsr-MuiTypography-root"
    >
      <b>
        Well this is awkward...
      </b>
    </p>
    <br />
    <p
      class="MuiTypography-root MuiTypography-body1 css-19f7wsr-MuiTypography-root"
      data-testid="end-of-feed-message"
    >
      You've reached the end. There's nothing more to see here
    </p>
  </div>
</DocumentFragment>
`;
